File: Python/2018/1a.py
import math

interest, repayment_percent = (int(x) for x in input('> ').split(" "))

debt = 10000
total_repayment = 0

while debt > 0:
	debt += math.ceil((interest / 100) * debt)
	repayment = math.ceil((repayment_percent/100) * debt)
	repayment = min(max(repayment, 5000), debt)
	debt -= repayment
	total_repayment += repayment


print(round(total_repayment/100, 2))

File: C++/2018/1a.cpp
#include <bits/stdc++.h>

using namespace std;

typedef unsigned int uint;
typedef long long ll;

#define F first
#define S second

const int INF = INT_MAX;

float float_fix(float n){
    // Tries to fix any floating-point errors that may occur
    n = (int)(n*100);
    return n/100;
}

int main() {
    float interest, repayment; cin >> interest >> repayment;
    interest /= 100; repayment /= 100;

    int debt = 10000;
    float paid = 0;
    while (debt > 0){
        debt += ceil(float_fix((float)debt * interest));
        int pay = ceil(float_fix((float)debt * repayment));
        if (pay < 5000) pay = 5000;
        if (pay > debt) pay = debt;
        paid += pay; debt -= pay;
    }
   printf("%.2f", paid/100);
}